packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -249,21 +249,25 @@ class AdaptiveScaffold extends StatefulWidget {
   /// the [NavigationRail] and [extended] for whether the [NavigationRail]
   /// is extended or not.
   static Builder standardNavigationRail({
+    required BuildContext context,
     required List<NavigationRailDestination> destinations,
     double width = 72,
     int? selectedIndex,
     bool extended = false,
-    Color backgroundColor = Colors.transparent,
+    Color? backgroundColor,
     EdgeInsetsGeometry padding = const EdgeInsets.all(8.0),
     Widget? leading,
     Widget? trailing,
     Function(int)? onDestinationSelected,
-    IconThemeData selectedIconTheme = const IconThemeData(color: Colors.black),
-    IconThemeData unselectedIconTheme =
-        const IconThemeData(color: Colors.black),
-    TextStyle selectedLabelTextStyle = const TextStyle(color: Colors.black),
+    IconThemeData? selectedIconTheme,
+    IconThemeData? unselectedIconTheme,
+    TextStyle? selectedLabelTextStyle,
+    TextStyle? unSelectedLabelTextStyle,
     NavigationRailLabelType labelType = NavigationRailLabelType.none,
   }) {
+    final NavigationRailThemeData navRailTheme =
```

Comment:
Why can't this logic be inside of the returned builder? It's odd to pass a `context` to this function, and if the logic can be in the returned builder, that makes more sense, since the context is then going to rebuild that context based on theme changes, not the context that was passed in.

Reply:
Moved logic to returned builder as suggested.